invenio_jobs/assets/semantic-ui/js/invenio_jobs/administration/RunsLogs.js

```diff
@@ -0,0 +1,212 @@
+  }
+
+  checkRunStatus = async (runId, jobId) => {
+    try {
+      this.cancellableFetch = withCancel(
+        http.get(`/api/jobs/${jobId}/runs/${runId}`)
+      );
+      const response = await this.cancellableFetch.promise;
+      if (response.status !== 200) {
+        throw new Error(`Failed to fetch run status: ${response.statusText}`);
+      }
+
+      const run = response.data;
+      run.formatted_started_at = this.formatDatetime(run.started_at);
+      const runDuration = this.getDurationInMinutes(run.started_at, run.finished_at);
+      this.setState({ run: run, runDuration: runDuration });
+      if (run.status === "SUCCESS" || run.status === "FAILED" || run.status === "PARTIAL_SUCCESS") {
+        clearInterval(this.logsInterval); // Stop fetching logs if run finihsed
+      }
+    } catch (err) {
+      console.error("Error checking run status:", err);
+      this.setState({ error: err.message });
+    }
+  };
+
+  componentDidMount() {
+    this.logsInterval = setInterval(async () => {
+      const { run, sort } = this.state;
+      if (run.status === "RUNNING") {
+        await this.fetchLogs(run.id, sort); // Fetch logs only if the run is running
+        await this.checkRunStatus(run.id, run.job_id); // Check the run status
+      }
+    }, 2000);
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.logsInterval);
+  }
+
+  render() {
+    const { error, logs, run, runDuration, sort } = this.state;
+    const levelClassMapping = {
+      DEBUG: "",
+      INFO: "primary",
+      WARNING: "warning",
+      ERROR: "negative",
+      CRITICAL: "negative",
+    };
+
+    const getClassForLogLevel = (level) => levelClassMapping[level] || "";
+    return (
+      <Container>
+      <Header as="h2" className="mt-20">
+        {run.title}
+      </Header>
+      <Divider />
+      {error && (
+        <Message negative>
+        <Message.Header>Error Fetching Logs</Message.Header>
+        <p>{error}</p>
+        </Message>
+      )}
+      <Grid celled>
+        <Grid.Row>
+        <Grid.Column width={3}>
+          <Header as="h4" color="grey">
+          Job run
+          </Header>
+          <List>
+          <List.Item>
+            <Icon
+            name={
+              run.status === "SUCCESS"
+              ? "check circle"
+              : run.status === "FAILED"
+                ? "times circle"
+                : run.status === "RUNNING"
+                ? "spinner"
+                : run.status === "PARTIAL_SUCCESS"
+                  ? "exclamation circle"
+                  : "clock outline"
+            }
+            color={
+              run.status === "SUCCESS"
+              ? "green"
+              : run.status === "FAILED"
+                ? "red"
+                : run.status === "RUNNING"
+                ? "blue"
+                : run.status === "PARTIAL_SUCCESS"
+                  ? "orange"
+                  : "grey"
+            }
```

Comment:
could you move this mapping up as class attr so it is not being recomputed on each render?
then you could only do here

```
name={self.uiStatusSerialiser[run.status]["icon"]}
color={self.uiStatusSerialiser[run.status]["color"]}
```

it becomes also reusable in line 171